Dockerfile

```diff
@@ -1,35 +1,56 @@
-# build stage
-FROM golang:1-alpine AS build-env
-RUN apk add --no-cache \
-    build-base \
-    linux-headers \
-    coreutils \
-    binutils \
-    libtool \
-    musl-dev \
-    git \
-    make \
-    gcc \
-    libc-dev \
-    zip \
-    ca-certificates \
-    libcap
-
-ENV GO111MODULE=on \
-    CGO_ENABLED=0
-    
-WORKDIR /src
+# get newest certificates
+FROM --platform=$BUILDPLATFORM alpine:3.16 AS ca-certs
+RUN apk add --no-cache ca-certificates
+RUN --mount=type=cache,target=/etc/ssl/certs \
+    update-ca-certificates 2>/dev/null || true
 
+# zig compiler
+FROM --platform=$BUILDPLATFORM ghcr.io/euantorano/zig:master AS zig-env
+
+# build environment
+FROM --platform=$BUILDPLATFORM golang:1-alpine AS build
+
+# required arguments(buildx will set target)
+ARG VERSION
+ARG BUILD_TIME
+ARG TARGETOS
+ARG TARGETARCH
+ARG TARGETVARIANT
```

Comment:
Didn't know about these, so makes sense to split and link the docs IMHO:
```suggestion
# required arguments
ARG VERSION
ARG BUILD_TIME

# auto provided by Docker
# https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope
ARG TARGETOS
ARG TARGETARCH
ARG TARGETVARIANT
```

Reply:
Sorry assumed it was common Docker knowledge.
Will add more comments.